### PR TITLE
Add set of user-assigned categories to receipt entity.

### PR DESCRIPTION
**Pull request by basaam0**
- Remove label field from receipt entity
- Add category field from request parameters
- Update upload unit tests

### CODE REVIEW

File: src/main/java/servlets/UploadReceiptServlet.java [hunk 1]
```diff
@@ -273,6 +280,13 @@ private String getBaseUrl(HttpServletRequest request) {
     return baseUrl;
   }
 
+  /**
+   * Gets the list of user-assigned categories from the request.
+   */
+  private Set<String> getCategories(HttpServletRequest request) {
+    return new HashSet<>(Arrays.asList(request.getParameterValues("categories")));
```
Comment (bryyeh) [inline src/main/java/servlets/UploadReceiptServlet.java:287]: I believe if there are duplicate categories, this will throw an exception. Is that what we want? If so, we may want to tell the user.

File: src/main/java/servlets/UploadReceiptServlet.java [hunk 2]
```diff
@@ -273,6 +280,13 @@ private String getBaseUrl(HttpServletRequest request) {
     return baseUrl;
   }
 
+  /**
+   * Gets the list of user-assigned categories from the request.
+   */
+  private Set<String> getCategories(HttpServletRequest request) {
```
Comment (bryyeh) [inline src/main/java/servlets/UploadReceiptServlet.java:286]: Use ImmutableSet.

As Effective Java Item 17 explains, "immutable classes are easier to design, implement, and use than mutable classes. They are less prone to error and are more secure."

File: src/main/java/servlets/UploadReceiptServlet.java (hunk 1 above)
Comment (basaam0) [reply src/main/java/servlets/UploadReceiptServlet.java]: I changed this method to use an ImmutableSet, so now it will remove duplicates. I have also added a unit test in #41 for duplicate categories.
